### PR TITLE
BUG: Interactive mode returned expression and not value in Jupyter

### LINKED ISSUE #2155: Interactive mode doesn't work in Jupyter
Running the next code in the terminal works as expected, `.count()` returns the number of rows as an `int`, since the option `interactive` is set to `True`:
```python
>>> import ibis
>>> print(ibis.__version__)
1.3.0+24.gd00a112.dirty
>>> ibis.options.interactive = True
>>> conn = ibis.sqlite.connect('geography.db')
>>> conn.table('countries').count()
252
```
But running the same exact code in a notebook returns the expression visualization instead:

ibis_interactive_bug

### PR DESCRIPTION
**Pull request by datapythonista**
Closes #2155

In Jupyter, interactive mode is not interactive, expressions are displayed.

I don't see `Expr` much tested, I guess not worth of having a test only for this.

### CODE REVIEW

**Comment by xmnlab** [comment]
@datapythonista maybe you can add a tests for that here: https://github.com/ibis-project/ibis/blob/master/ibis/expr/tests/test_interactive.py

let me know if it makes sense.

File: ibis/expr/tests/test_interactive.py [hunk 1]
```diff
@@ -30,6 +30,12 @@ def test_interactive_execute_on_repr(self):
 
         assert len(self.con.executed_queries) > 0
 
+    def test_png_repr_returns_correct_type(self):
+        table = self.con.table('functional_alltypes')
+
+        with config.option_context('interactive', True):
```
Comment (xmnlab) [inline ibis/expr/tests/test_interactive.py:36]: maybe would be good/useful to add here also a test for interactive False?

Comment (datapythonista) [reply ibis/expr/tests/test_interactive.py:36]: Thought on that, it's a good point. But that's trickier, as it depends on other factors like libraries that are installed. I think it's out of scope here.

Comment (xmnlab) [reply ibis/expr/tests/test_interactive.py:36]: ok sounds good to me!

**Comment by datapythonista** [comment]
@jreback do you mind having a look at this? It's been rebased and is green. I need this bug fixed to work on the tutorial.

File: ibis/expr/tests/test_interactive.py [hunk 2]
```diff
@@ -30,6 +30,12 @@ def test_interactive_execute_on_repr(self):
 
         assert len(self.con.executed_queries) > 0
 
+    def test_png_repr_returns_correct_type(self):
```
Comment (jreback) [inline ibis/expr/tests/test_interactive.py:33]: can you add a test when interactive is False a well

Comment (datapythonista) [reply ibis/expr/tests/test_interactive.py:33]: Good point. I tried that in the past, and it's not trivial. There are different cases that affect the output of `_repr_png_`. Like, graphviz not being installed, anything that makes importing `ibis.expr.visualize` fail, or call the `to_graph` method. Also if the `graphviz_repr` option is set to `False`.

I added the test, to see what fails, but I think it'll be tricky to find a way to add that test in a deterministic and reliable way. We may end up introducing a test that doesn't provide much value, but it's flaky when new builds are added.

Comment (jreback) [reply ibis/expr/tests/test_interactive.py:33]: right its just to verify that it is NOT none

Comment (datapythonista) [reply ibis/expr/tests/test_interactive.py:33]: Yep, but the problem is that even with `interactive` set to `False`, `_repr_png_` can still be `None` for different reasons:
- graphviz not installed
- option `graphviz_repr` set to `False`
- an exception happens when importing the visualization module

I added a test for the case when all conditions are met for the result to be `not None`. It's a bit tricky, but I think it's the best we can do.

File: ibis/expr/tests/test_interactive.py [hunk 3]
```diff
@@ -30,6 +30,24 @@ def test_interactive_execute_on_repr(self):
...
+        table = self.con.table('functional_alltypes')
+
+        with config.option_context('interactive', True):
+            assert table._repr_png_() is None
+
+        try:
+            import ibis.expr.visualize
```
Comment (jreback) [inline ibis/expr/tests/test_interactive.py:40]: this import doesn't need to be here

File: ibis/expr/tests/test_interactive.py [hunk 4]
```diff
@@ -30,6 +30,24 @@ def test_interactive_execute_on_repr(self):
 
         assert len(self.con.executed_queries) > 0
 
+    def test_png_repr_returns_correct_type(self):
+        table = self.con.table('functional_alltypes')
+
+        with config.option_context('interactive', True):
+            assert table._repr_png_() is None
+
+        try:
```
Comment (jreback) [inline ibis/expr/tests/test_interactive.py:39]: make this a new test

File: ibis/expr/tests/test_interactive.py [hunk 5]
```diff
@@ -30,6 +30,24 @@ def test_interactive_execute_on_repr(self):
...
+            import ibis.expr.visualize
+
+            ibis.expr.visualize.to_graph(table).pipe(format='png')
+        except Exception:
+            pass
```
Comment (jreback) [inline ibis/expr/tests/test_interactive.py:44]: here just skip the test if you can't format

**Comment by datapythonista** [comment]
Thanks for the feedback @jreback, it was useful. This should be ready now.

**Comment by jreback** [comment]
thanks @datapythonista

**Pull request merged**